educator/src/Dscp/Educator/Config.hs

```diff
@@ -1,30 +1,48 @@
+{-# LANGUAGE TypeOperators #-}
+
 -- | All educator's configurations.
 
 module Dscp.Educator.Config
     ( EducatorConfig
     , EducatorConfigRec
     , HasEducatorConfig
+    , educatorConfig
     , withEducatorConfig
     , fillEducatorConfig
 
     , module Dscp.Witness.Config
     ) where
 
-import Data.Reflection (Given)
-import Loot.Config (ConfigKind (Final, Partial), ConfigRec)
+import Data.Reflection (Given, give, given)
+import Loot.Config ((:::), (::<), ConfigKind (Final, Partial), ConfigRec)
 
+import Dscp.Config
+import Dscp.DB.SQLite (SQLiteParams)
+import Dscp.Educator.Launcher.Params (EducatorKeyParams)
+import Dscp.Educator.Web.Params (EducatorWebParams)
 import Dscp.Witness.Config
 
-
-type EducatorConfig = CoreConfig
+type EducatorConfig = WitnessConfig +++
+    '[ "educator" ::<
+       '[ "db" ::: SQLiteParams
+        , "keys" ::: EducatorKeyParams
+        , "api" ::: EducatorWebParams
+        ]
+     ]
```

Comment:
Total configuration occurs to be ambiguous as soon as some parameters are present both in `witnessConfig` and `educatorConfig`.
Could we make `EducatorConfig` contain no fields of `WitnessConfig`; then parse `WitnessConfig ++ EducatorConfig` and `rcast` it appropriately when passing to `withWitnessConfig` and `withEducatorConfig`?

Reply:
@Martoon-00 Well, I think you're wrong, because currently configuration doesn't contain any duplicated data (similar `api` and `keys` fields in `witness` and `educator` subsections are about different keys and different APIs). Though your suggestion is legit anyway; I even wanted to separate config further, having parts like `LoggingConfig`, `NetworkConfig`, etc.